src/AltinnCore/Runtime/RestControllers/ProcessController.cs

```diff
@@ -41,29 +66,445 @@ public class ProcessController : ControllerBase
+
+                if (nextElements.Count > 1)
+                {
+                    return Conflict($"Cannot complete process. Multiple outgoing sequence flows detected from task {currentTaskId}. Please select manually among {nextElements}");
+                }
+
+                string nextElement = nextElements.First();
+
+                if (InstanceIsValid(instance))
+                {
+                    instance = await UpdateProcessStateToNextElement(org, app, instance, instanceOwnerId, instanceGuid, nextElement);                    
+                }
+                else
+                {
+                    return Conflict($"Instance is not valid in task {currentTaskId}. Automatic completeion of process is stopped");
+                }
+
+                currentTaskId = instance.Process.CurrentTask?.ElementId;
+            }
+            while (instance.Process.EndEvent == null || counter > MAX_ITERATIONS_ALLOWED);
+
+            if (counter > 1000)
+            {
+                return StatusCode(500, $"More than {counter} iterations detected in process. Possible loop. Fix app process definition!");
+            }
+
+            return Ok(instance.Process);
+        }
+
+        private async Task<Instance> StartProcessOfInstance(string org, string app, int instanceOwnerId, Guid instanceGuid, Instance instance, string validStartElement)
+        {
+            DateTime now = DateTime.UtcNow;
+
+            instance.Process = new ProcessState
+            {
+                Started = now,
+                StartEvent = validStartElement,
+            };
+            Instance updatedInstance = await instanceService.UpdateInstance(instance, app, org, instanceOwnerId, instanceGuid);
+            List<InstanceEvent> events = new List<InstanceEvent>
+            {
+                GenerateProcessChangeEvent("process:StartEvent", updatedInstance, now)
+            };
+
+            await DispatchEvents(org, app, events);
+
+            return updatedInstance;
+        }
+
+        private async Task<Instance> UpdateProcessStateToNextElement(string org, string app, Instance instance, int instanceOwnerId, Guid instanceGuid, string nextElementId)
+        {
+            List<InstanceEvent> events = ChangeProcessState(instance, nextElementId);
+
+            Instance changedInstance = await instanceService.UpdateInstance(instance, app, org, instanceOwnerId, instanceGuid);
+
+            await DispatchEvents(org, app, events);
+
+            return changedInstance;
+        }
+
+        private async Task DispatchEvents(string org, string app, List<InstanceEvent> events)
+        {
+            foreach (InstanceEvent instanceEvent in events)
+            {
+                await eventService.SaveInstanceEvent(instanceEvent, org, app);
+            }
+        }
+
+        private string CheckOrGetValidNextElement(string currentElementId, string proposedElementId, out ActionResult nextElementError)
```

Comment:
The function name is somewhat confusing.. I think I prefere the 'GetValidStartEvenOrError'-format which was previously used. Could this function name be based on the same structure?